### PR TITLE
Issue #13328: Kill mutation for OneStatementPerLineCheck

### PR DESCRIPTION
**Pull request by Kevin222004**
Issue #13328: Kill mutation for OneStatementPerLineCheck

-------------

# Check :-
 https://checkstyle.org/config_coding.html#OneStatementPerLine

-----------

# Mutation covered 
https://github.com/checkstyle/checkstyle/blob/6324a9d693324eb755c2cdfdbd90c2300d3c7f02/config/pitest-suppressions/pitest-coding-2-suppressions.xml#L174-L181

### CODE REVIEW

**Comment by romani** [comment]
You need to provide long explanation why not possible to cover by Inputs.

We do not accept testing by pure unit test.

**Review by romani — CHANGES_REQUESTED** [review]
See above

**Comment by Kevin222004** [comment]
@romani done input based test added

**Review by Vyom-Yadav — APPROVED** [review]
LGTM!

**Comment by romani** [comment]
@Kevin222004 , please resolve conflict and look at CI failing

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/OneStatementPerLineCheckTest.java [hunk 1]
```diff
@@ -139,4 +143,20 @@ public void testAllTheCodeInSingleLine() throws Exception {
         verify(checkConfig, getPath("InputOneStatementPerLine.java"),
                 expected);
     }
+
+    @Test
+    public void testStateIsClearedOnBeginTree1() throws Exception {
```
[the remark this answers is not in the document]
Comment (Kevin222004) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/OneStatementPerLineCheckTest.java]: Done

**Review by romani — APPROVED** [review]
Ok to merge if CI is green

**Comment by rdiachenko** [comment]
@Kevin222004 please resolve a conflict

**Review by rdiachenko — APPROVED** [review]
lgtm

**Comment by rdiachenko** [comment]
rebased via Github

**Pull request merged**